src/web/components/Core/Form.tsx

```diff
@@ -1,23 +1,33 @@
 import axios from 'axios';
-import React, { cloneElement, isValidElement, ReactElement, ReactNode } from 'react';
-import { DeepPartial, FieldValues, SubmitHandler, useForm } from 'react-hook-form';
+import React, { createContext, ReactNode, useCallback, useMemo } from 'react';
+import { Control, DeepPartial, FieldValues, SubmitHandler, useForm } from 'react-hook-form';
 
 import './Form.scss';
 
 type FormProps<T extends FieldValues> = {
   onSubmit: SubmitHandler<T>;
-  children: ReactNode[];
+  children: ReactNode;
   onError?: (error: unknown) => void;
   defaultValues?: DeepPartial<T>;
   submitButtonText?: string;
+  customizeSubmit?: boolean;
 };
 
+export type FormContextType<T extends FieldValues> = {
+  control: Control<T>;
+  handleSubmit: (e?: React.BaseSyntheticEvent) => Promise<void>;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const FormContext = createContext<FormContextType<any> | null>(null);
```

Comment:
What did you try to get rid of the any here?

Reply:
As the generic type, `T extends FieldValues` is not defined outside of the Form component, I tried to use just `FieldValues` but I get the error:

Type '{ control: Control<T, any>; handleSubmit: (e?: React.BaseSyntheticEvent<object, any, any> | undefined) => Promise<void>; }' is not assignable to type 'FormContextType<FieldValues>'.